### PR TITLE
Pool filters frontend

### LINKED ISSUE #531: [Pool Filter] Frontend of adding pool filters to the Pools page
* Change the top navigation from seaweed & indigo -> white & black

* Add pool filter tabs on top of all pools. Tabs: All, BTC, USD, ETH, OUTDATED

* Add filter logic to pool setting

### PR DESCRIPTION
**Pull request by sandrasong**
**Pool order is updated**

Screen Shot 2021-07-14 at 8 42 01 PM

07:14:2021_pool_filters

Close #531

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "saddle-frontend",
```
Comment (alphastorm) [inline package-lock.json:8]: why was this removed?

File: src/pages/Pools.tsx [hunk 2]
```diff
@@ -5,49 +5,113 @@ import {
...
-          poolData={usdPoolData}
-          poolRoute={`/pools/usd`}
-          userShareData={usdUserShareData}
-        />
-        <PoolOverview
-          poolData={veth2PoolData}
-          poolRoute={`/pools/veth2`}
-          userShareData={veth2UserShareData}
-        />
+        {Object.values(POOLS_MAP)
```
Comment (alphastorm) [inline src/pages/Pools.tsx:105]: this places the alETH and D4 pools at the very bottom, can we update it to preserve the current ordering (sorted by TVL)

**Review by alphastorm — COMMENTED** [review]
lgtm other than a couple of minor comments! 🤠

File: src/pages/Pools.tsx [hunk 3]
```diff
@@ -5,49 +5,113 @@ import {
...
+        userShareData: veth2UserShareData,
+        poolRoute: "/pools/veth2",
+      }
+    }
+  }
   return (
     <div className={styles.poolsPage}>
       <TopMenu activeTab="pools" />
+      <ul className={styles.filters}>
+        <li
```
Comment (hammeiam) [inline src/pages/Pools.tsx:62]: You can keep this DRY by mapping over a list
```
[
  ["all", "ALL"],
  [PoolTypes.BTC, "BTC"],
  [PoolTypes.ETH, "ETH"],
  [PoolTypes.USD, "USD"],
  ["outdated", "OUTDATED"],
].map(([filterKey, text]) => (
  <li
    key={filterKey}
    className={classNames(styles.filterTab, {
      [styles.selected]: filter === filterKey,
    })}
    onClick={(): void => setFilter(filterKey)}
  >
{displayText}
</li>
))
```

File: src/pages/Pools.tsx (hunk 2 above)
Comment (hammeiam) [reply src/pages/Pools.tsx:105]:
```
{
Object.values(POOLS_MAP)
.filter(...)
.map(({ name }) => getPropsForPool(name))
.sort((a,b) => {
  // 1. active pools 
  // 2. user pools
  // 3. higher TVL pools
  if (a.poolData?.migrate != null) {
    return 1
  } else if (b.poolData?.migrate != null) {
    return -1
  } else if (a.userShareData?.usdBalance) {
    if (b.userShareData?.usdBalance) {
      return a.userShareData.usdBalance.gt(b.userShareData.usdBalance) ? -1 : 1
    }
    return -1
  } else if (a.poolData?.reserve) {
    if (b.poolData?.reserve) {
      return a.poolData.reserve.gt(b.poolData.reserve) ? -1 : 1
    }
    return -1
  }
  return -1
}) 
}
.map(poolProps => <PoolOverview key={poolProps.name} {...poolProps} />) // TODO add name to poolProps
```

File: src/styles/theme.scss [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
 $primary: #00f4d7;
 $primary-blue: #3800d6;
 $primary-blue-light: #f5f2fd;
 $primary-blue-dark: #0d0033;
 $white: #ffffff;
-$black: #000;
+$black: #000000;
 $error: #fb5a5a;
-$black80: #333333;
+$black80: #{$black}cc;
```
Comment (hammeiam) [inline src/styles/theme.scss:12]: can we not use alpha for colors? Also colors with alpha should be labeled as such eg `$blue40T` (but again, lets not use alpha where we can avoid it)

**Review by alphastorm — APPROVED** [review]

File: src/pages/Pools.tsx [hunk 5]
```diff
@@ -5,49 +5,124 @@ import {
...
+                ? -1
+                : 1
+            } else {
+              return (a.poolData?.reserve || Zero).gt(
+                b.poolData?.reserve || Zero,
+              )
+                ? -1
+                : 1
+            }
+            return -1
```
Comment (hammeiam) [inline src/pages/Pools.tsx:121]: Can delete, this should be unreachable

**Review by hammeiam — APPROVED** [review]
Looks good to me, great work!
